aexpect/remote_door.py

```diff
@@ -174,13 +174,20 @@ def run_remotely(function):
     and is expected to follow PEP8 standards.
     """
     def wrapper(session, *args, **kwargs):
+        # drop first argument and first line with the decorator since function already runs remotely
+        fn_source = inspect.getsourcelines(function)[0][1:]
+        indent_number = len(fn_source[0]) - len(fn_source[0].lstrip())
+
+        # remove extra indentation from the beginning to allow indented functions
+        if indent_number > 0:
+            fn_source = [line[indent_number:] for line in fn_source if len(line) > indent_number]
+
         wrapper_control = "import logging\n"
         wrapper_control += "logging.basicConfig(level=logging.DEBUG, format='%(module)-16.16s '\n"  # pylint: disable=C0301
         wrapper_control += "                    'L%(lineno)-.4d %(levelname)-5.5s| %(message)s')\n"  # pylint: disable=C0301
         wrapper_control += "import sys\n"
         wrapper_control += "sys.path.append('%s')\n" % REMOTE_PYTHON_PATH
-        # drop first argument and first line with the decorator since function already runs remotely
-        wrapper_control += "\n" + "\n".join(inspect.getsource(function).split("\n")[1:]).replace("_session, ", "")  # pylint: disable=C0301
+        wrapper_control += "\n" + "".join(fn_source).replace("_session, ", "")
```

Comment:
it'd be probably nice to expand the selftest. How about this:

```diff
diff --git a/tests/test_remote_door.py b/tests/test_remote_door.py
index 07facee..297b2fb 100644
--- a/tests/test_remote_door.py
+++ b/tests/test_remote_door.py
@@ -23,12 +23,6 @@ import unittest.mock as mock
 from aexpect import remote_door
 
 
-@remote_door.run_remotely
-def add_one(number):
-    """A small decorated test function."""
-    return number + 1
-
-
 @mock.patch('aexpect.remote_door.remote', mock.MagicMock())
 class RemoteDoorTest(unittest.TestCase):
     """Unit test class for the remote door."""
@@ -84,6 +78,14 @@ class RemoteDoorTest(unittest.TestCase):
 
     def test_run_remote_decorator(self):
         """Test that a remote utility object can be run properly."""
+        @remote_door.run_remotely
+        def add_one(number):
+            """A small decorated test function."""
+            def foo():
+
+                pass
+
+            return number + 1
         _ = add_one(self.session, 3)  # pylint: disable=E1121
         # python 3.6 and above
         if hasattr(self.session.cmd, "assert_called_once"):
@@ -99,6 +101,8 @@ class RemoteDoorTest(unittest.TestCase):
             self.assertIn("def add_one(number):\n", control_lines)
             self.assertIn("    return number + 1\n", control_lines)
             self.assertIn("result = add_one(3)\n", control_lines)
+            self.assertIn("    def foo():\n", control_lines)
+            self.assertIn("        pass\n", control_lines)
 
     def test_run_remote_object(self):
         """Test that a remote utility object can be run properly."""
```

Reply:
Good combination of localizing dependency functions like `add_one` and including indentation into the testing. All done.